src/ui/components/Library/Team/View/Tests/hooks/useTests.ts

```diff
@@ -24,3 +30,92 @@ export function useTests(): Test[] {
   );
   return value;
 }
+
+export function useTest(testId: string) {
+  const { teamId } = useContext(TeamContext);
+  const { data, loading } = useQuery(GET_TEST, {
+    variables: { workspaceId: teamId, testId },
+  });
+
+  if (loading) {
```

Comment:
:hand: This whole section should be wrapped in `useMemo(cb, [data, loading])` so we only recalculate it when this changes.